PF/Assignment1/BikeJdbcHelper.java

```diff
@@ -0,0 +1,138 @@
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+public class BikeJdbcHelper {
+
+	// function to create bike table
+	public static void createBikeTable() throws VehicleManagementException {
+		Connection con = null;
+		Statement stmt = null;
+		ConnectionUtil conUtil = new ConnectionUtil();
+		/* creates connection to db */
+		con = conUtil.getConnection();
+
+		String query = "CREATE TABLE IF NOT EXISTS Bike"
+				+ "("
+				+ "bike_id INT AUTO_INCREMENT PRIMARY KEY,"
+				+ "self_start tinyint(1),"
+				+ "helmet_price DOUBLE,"
+				+ "vehicle_id INT,"
+				+ "CONSTRAINT `vehicle_key1` FOREIGN KEY (vehicle_id) REFERENCES Vehicle (vehicle_id) ON DELETE CASCADE"
+				+ ")";
+		try {
+			stmt = con.createStatement();
+			stmt.executeUpdate(query);
+
+		} catch (SQLException e) {
+			e.printStackTrace();
+		} finally {
+			/* close connection */
+			try {
+				if (con != null) {
+					con.close();
+				}
+				if (stmt != null) {
+					stmt.close();
+				}
+
+			} catch (SQLException e) {
+				e.printStackTrace();
+			}
+		}
+	}
+
+	// function to insert in bike
+	public static int insertInBikeTable(Bike objBike)
+			throws VehicleManagementException {
+		int countBike = 0;
+		Connection con = null;
+		PreparedStatement ps = null;
+		ConnectionUtil conUtil = new ConnectionUtil();
+		/* creates connection to db */
+		con = conUtil.getConnection();
+
+		String query = "INSERT INTO Bike ( self_start, helmet_price, vehicle_id ) VALUES (?,?,?)";
+
+		try {
+			ps = (PreparedStatement) con.prepareStatement(query);
+			/* set variable in prepared statement */
+			ps.setBoolean(1, objBike.getSelfStart());
+			ps.setDouble(2, objBike.getHelmetPrice());
+			ps.setInt(3, objBike.getVehicleId());
+			countBike = ps.executeUpdate();
+		} catch (SQLException e) {
+			e.printStackTrace();
+		} finally {
+			/* close connection */
+			try {
+				if (con != null) {
+					con.close();
+				}
+				if (ps != null) {
+					ps.close();
+				}
+			} catch (SQLException e) {
+				e.printStackTrace();
+			}
+		}
+		return countBike;
+	}
+
+	// Function to delete from bike
+	public static void deleteBikeTable(int vehicleId)
+			throws VehicleManagementException {
+		Connection con = null;
+		Statement stmt = null;
+		ResultSet rs = null;
+		ConnectionUtil conUtil = new ConnectionUtil();
+		/* creates connection to db */
+		con = conUtil.getConnection();
+
+		String query = "SELECT vehicle_id FROM Car WHERE vehicle_id = "
```

Comment:
ACTION: Do we required to check first record is exists? if we directly execute delete query it will return number of rows effected, if it is 0 then record was not present in that table and if it is greater than 0 then records deleted successfully.